dask_jobqueue/core.py

```diff
@@ -363,33 +361,84 @@ def stop_workers(self, workers):
                 jobs.append(_job_id_from_worker_name(w['name']))
             else:
                 jobs.append(_job_id_from_worker_name(w.name))
-        self.stop_jobs(set(jobs))
+        self.stop_jobs(jobs)
 
     def stop_jobs(self, jobs):
         """ Stop a list of jobs"""
-        logger.debug("Stopping jobs: %s" % jobs)
+        logger.debug("Stopping jobs: %s", jobs)
         if jobs:
             jobs = list(jobs)
             self._call([self.cancel_command] + list(set(jobs)))
 
+        # if any of these jobs were pending, we should remove those now
+        for job_id in jobs:
+            if job_id in self.pending_jobs:
+                del self.pending_jobs[job_id]
+
+    def scale(self, n):
+        """ Scale cluster to n workers
+        Parameters
+        ----------
+        n: int
+            Target number of workers
+        Example
+        -------
+        >>> cluster.scale(10)  # scale cluster to ten workers
+        See Also
+        --------
+        Cluster.scale_up
+        Cluster.scale_down
+        """
+        with log_errors():
+            active_and_pending = self._count_active_and_pending_workers()
+            if n >= active_and_pending:
```

Comment:
these two lines are why I am currently overloading the scale method. We need to know if we want to scale up or scale down and that depends on the number of pending workers. The current `Cluster.scale()` method does not have a way to evaluate pending workers.

Reply:
This part is currently done in `scale_up`, maybe not correctly, but what prevents us from doing it there?

Something like this:
````python
    def scale_up(self, n, **kwargs):
            """ Brings total worker count up to ``n`` """
            active_and_pending = self._count_active_and_pending_workers()
            if n >= active_and_pending:
                 self.start_workers(n - self._count_active_and_pending_workers())
            else:
                n_to_close = active_and_pending - n
                if n_to_close < self._count_pending_workers():
                    # We only need to kill some pending jobs
                    to_kill = int(n_to_close / self.worker_processes)
                    jobs = list(self.pending_jobs.keys())[to_kill:]
                    self.stop_jobs(jobs)
                else:
                    raise something, we should never be there in scale_up.
````